docs(releases): stellar GMP v1.0.0 and ITS v1.0.0 (#523)
axelarnetwork/axelar-contract-deployments: releases/cosmwasm/2025-01-Stellar-GMP-v1.0.0.md

| **Deployment** | @RiceAndMeet <steven@interoplabs.io> |

| **Network**          | **Deployment Status** | **Date** |
| -------------------- | --------------------- | -------- |
| **Devnet Amplifier** | -                     | TBD      |
| **Stagenet**         | -                     | TBD      |
| **Testnet**          | -                     | TBD      |
| **Mainnet**          | -                     | TBD      |

- [Amplifier Releases](https://github.com/axelarnetwork/axelar-amplifier/releases)
- [VotingVerifier v1.1.0](https://github.com/axelarnetwork/axelar-amplifier/releases/tag/voting-verifier-v1.1.0)
- [Gateway v1.1.1](https://github.com/axelarnetwork/axelar-amplifier/releases/tag/gateway-v1.1.1)
- [MultisigProver v1.1.1](https://github.com/axelarnetwork/axelar-amplifier/releases/tag/multisig-prover-v1.1.1)

## Background

1. These are the instructions for deploying Amplifier contracts for Stellar connection.

### Pre-requisites

- Ensure that the [External Gateway](../stellar/2025-01-Stellar-GMP-v1.0.0.md) is deployed first, as `VotingVerifier` needs the `sourceGatewayAddress` which is the External Gateway address.

## Deployment

> - Create an `.env` config. `CHAIN` should be set to `stellar` for mainnet, and `stellar-2024-q4` for all other networks.

Reviewer: perhaps?
```suggestion
- Create an `.env` config. `CHAIN` should be set to `stellar` for mainnet, and `stellar-2025-q1` for all other networks.
```

Author: The chain name has been discussed, and decided to use this format: `stellar-[stellar testnet release]-[optional counter only if needed]` for testnet, and mainnet: `stellar`
reference: https://github.com/axelarnetwork/axelar-contract-deployments/pull/527#discussion_r1940239861

```yaml
MNEMONIC=xyz
ENV=xyz
CHAIN=xyz
```

- Confirm `VotingVerifier(v1.1.0)`, `Gateway(v1.1.1)` and `MultisigProver(v1.1.1)` contracts are already stored in `$ENV.json`

```bash
VotingVerifier(v1.1.0) -> "storeCodeProposalCodeHash": "d9412440820a51bc48bf41a77ae39cfb33101ddc6562323845627ea2042bf708"
Gateway(v1.1.1) -> "storeCodeProposalCodeHash": "2ba600ee0d162184c9387eaf6fad655f1d75db548f93e379f0565cb2042d856f"
MultisigProver(v1.1.1) -> "storeCodeProposalCodeHash": "00428ef0483f103a6e1a5853c4b29466a83e5b180cc53a00d1ff9d022bc2f03a"
```

- Add config in `$ENV.json` to deploy Amplifier contracts.

| Network              | `serviceName` | `votingThreshold` | `signingThreshold` | `confirmationHeight` |
| -------------------- | ------------- | ----------------- | ------------------ | -------------------- |
| **Devnet-amplifier** | `validators`  | `["6", "10"]`     | `["6", "10"]`      | `1`                  |
| **Stagenet**         | `amplifier`   | `["51", "100"]`   | `["51", "100"]`    | `1`                  |
| **Testnet**          | `amplifier`   | `["51", "100"]`   | `["51", "100"]`    | `1`                  |

> | **Mainnet**          | `amplifier`   | `["2", "3"]`      | `["2", "3"]`       | 1                |

Reviewer: ```suggestion
| **Mainnet**          | `amplifier`   | `["2", "3"]`      | `["2", "3"]`       | `1`               |
```

Author: updaated

REWARDS=$(cat ./axelar-chains-config/info/$ENV.json | jq .axelar.contracts.Rewards.address | tr -d '"')
```

- Gov proposal environment variables. Update these for each network

```bash
RUN_AS_ACCOUNT=[wasm deployer key address]
PROVER_ADMIN=[prover admin]
DEPOSIT_VALUE=100000000
REWARD_AMOUNT=1000000uamplifier
```

5. Register stellar gateway at the Router

```bash
node cosmwasm/submit-proposal.js execute \
  -c Router \
  -t "Register Gateway for stellar" \
  -d "Register Gateway address for stellar at Router contract" \
  --deposit $DEPOSIT_VALUE \
  --runAs $RUN_AS_ACCOUNT \
  --msg "{
    \"register_chain\": {
      \"chain\": \"$CHAIN\",
      \"gateway_address\": \"$GATEWAY\",
      \"msg_id_format\": \"hex_tx_hash_and_event_index\"
      }
    }"
```

6. Register chain and public key from ampd. Ask verifiers to run these.

```bash
for i in $(seq 0 4); do kubectl exec -it ampd-set-2-axelar-amplifier-worker-"$i" -n $ENV -c ampd -- ampd register-chain-support "[service name]" $CHAIN ; done

for i in $(seq 0 4); do kubectl exec -it ampd-set-2-axelar-amplifier-worker-"$i" -n $ENV -c ampd -- ampd register-public-key ed25519 ; done

> ```

Reviewer: I think it would be fine to format/ident this for readability

Author: updated

# Stellar GMP Amplifier

|                | **Owner**                            |
| -------------- | ------------------------------------ |
| **Created By** | @ahramy <ahram@interoplabs.io>       |
| **Deployment** | @RiceAndMeet <steven@interoplabs.io> |

| **Network**          | **Deployment Status** | **Date** |
| -------------------- | --------------------- | -------- |
| **Devnet Amplifier** | -                     | TBD      |
| **Stagenet**         | -                     | TBD      |
| **Testnet**          | -                     | TBD      |
| **Mainnet**          | -                     | TBD      |

- [Amplifier Releases](https://github.com/axelarnetwork/axelar-amplifier/releases)
- [VotingVerifier v1.1.0](https://github.com/axelarnetwork/axelar-amplifier/releases/tag/voting-verifier-v1.1.0)
- [Gateway v1.1.1](https://github.com/axelarnetwork/axelar-amplifier/releases/tag/gateway-v1.1.1)

> - [MultisigProver v1.1.1](https://github.com/axelarnetwork/axelar-amplifier/releases/tag/multisig-prover-v1.1.1)

Author: need to check with @RiceAndMeet

> - [MultisigProver v1.1.1](https://github.com/axelarnetwork/axelar-amplifier/releases/tag/multisig-prover-v1.1.1)

Author: double checked with Steven. resolving

| -------------------- | ----------------- | ------------------ |
| **Devnet-amplifier** | `["6", "10"]`     | `["6", "10"]`      |
| **Stagenet**         | `["6", "10"]`     | `["6", "10"]`      |
| **Testnet**          | `["6", "10"]`     | `["6", "10"]`      |
| **Mainnet**          | `TBD`             | `TBD`              |

```bash
# Add config under "axelar": "contracts": "VotingVerifier" based on Network
\"$CHAIN\" : {
  "governanceAddress": "[governance address]",
  "serviceName": "validators",
  "sourceGatewayAddress": "[external gateway address]",
  "votingThreshold": [
    "6",
    "10"
  ],
  "blockExpiry": 10,
  "confirmationHeight": 1,
  "msgIdFormat": "hex_tx_hash_and_event_index",
  "addressFormat": "stellar"
}

# Add config under "axelar": "contracts": "MultisigProver" based on Network
\"$CHAIN\" : {
  "governanceAddress": "[governance address]]",
  "adminAddress": "[admin address]",
  "signingThreshold": [
    "6",
    "10"
  ],
  "serviceName": "validators",
  "verifierSetDiffThreshold": 0,
  "encoder": "stellar_xdr",
  "keyType": "ed25519"
}
```

### Deploy Amplifier contracts that connect to the Stellar gateway.

1. Instantiate `VotingVerifier`

```bash

> node ./cosmwasm/deploy-contract.js instantiate -c VotingVerifier --fetchCodeId --instantiate2

Author: I checked with Edwin, it needs to have `--admin` option to make sure setting up the contract admin, i.e. ```
node ./cosmwasm/deploy-contract.js instantiate -c VotingVerifier --fetchCodeId --instantiate2 --admin [wasm contract admin address for the upgrade/migration]
```
updated.

> node ./cosmwasm/deploy-contract.js instantiate -c VotingVerifier --fetchCodeId --instantiate2

Author: updated

GATEWAY=$(cat ./axelar-chains-config/info/$DEVNET.json | jq ".axelar.contracts.Gateway[\"$CHAIN\"].address" | tr -d '"')
MULTISIG_PROVER=$(cat ./axelar-chains-config/info/$DEVNET.json | jq ".axelar.contracts.MultisigProver[\"$CHAIN\"].address" | tr -d '"')
MULTISIG=$(cat ./axelar-chains-config/info/$DEVNET.json | jq .axelar.contracts.Multisig.address | tr -d '"')
REWARDS=$(cat ./axelar-chains-config/info/$DEVNET.json | jq .axelar.contracts.Rewards.address | tr -d '"')
```

- General environment variables

```bash
RUN_AS_ACCOUNT=[wasm deployer key address]
DEPOSIT_VALUE=100000000
REWARD_AMOUNT=1000000uamplifier
```

5. Register stellar gateway at the Router

```bash
node cosmwasm/submit-proposal.js execute \
  -c Router \
  -t "Register Gateway for stellar" \
  -d "Register Gateway address for stellar at Router contract" \
  --deposit $DEPOSIT_VALUE \
  --runAs $RUN_AS_ACCOUNT \
  --msg "{
    \"register_chain\": {
      \"chain\": \"$CHAIN\",
      \"gateway_address\": \"$GATEWAY\",
      \"msg_id_format\": \"hex_tx_hash_and_event_index\"
      }
    }"
```

> 6. Register chain on ampd

Reviewer 2: ```suggestion
6. Register chain on ampd. Ask verifiers to run these.
```

> 6. Register chain on ampd

Reviewer 2: Ordering is weird, run this after the proposals are submitted, since validators need to vote on those, along with running ampd cmds

Author: updated ordering, please let me know if it's correct

```bash

> for i in $(seq 0 4); do kubectl exec -it ampd-set-2-axelar-amplifier-worker-"$i" -n $ENV -c ampd -- ampd register-chain-support validators $CHAIN ; done

Reviewer 2: service name is not always `validators`

for i in $(seq 0 4); do kubectl exec -it ampd-set-2-axelar-amplifier-worker-"$i" -n $ENV -c ampd -- ampd register-public-key ed25519 ; done

> ```

Reviewer 2: Again, avoid including kubectl cmds. Document the actual ampd cmd that a verifier needs to run

Author: I will double check with @RiceAndMeet and will update

Reviewer 2: The ampd cmds just need to be shown separately, e.g.
`ampd register-public-key ed25519`

7. Register prover contract on coordinator

```bash
node cosmwasm/submit-proposal.js execute \
  -c Coordinator \
  -t "Register Multisig Prover for stellar" \
  -d "Register Multisig Prover address for stellar at Coordinator contract" \
  --deposit $DEPOSIT_VALUE \
  --runAs $RUN_AS_ACCOUNT \
  --msg "{
    \"register_prover_contract\": {
      \"chain_name\": \"$CHAIN\",
      \"new_prover_addr\": \"$MULTISIG_PROVER\"
    }
  }"
```

8. Authorize Stellar Multisig prover on Multisig

```bash
node cosmwasm/submit-proposal.js execute \
  -c Multisig \
  -t "Authorize Multisig Prover for stellar" \
  -d "Authorize Multisig Prover address for stellar at Multisig contract" \
  --runAs $RUN_AS_ACCOUNT \
  --deposit $DEPOSIT_VALUE \
  --msg "{
    \"authorize_callers\": {
      \"contracts\": {
        \"$MULTISIG_PROVER\": \"$CHAIN\"
      }
    }
  }"
```

> 9. Update verifier set

Reviewer 2: ```suggestion
9. Create genesis verifier set
```

> 9. Update verifier set

Reviewer 2: This step can only be run once sufficient verifiers have registered. Should indicate that

Author: Added note

```bash

> axelard tx wasm execute $MULTISIG_PROVER '"update_verifier_set"' --from amplifier --gas auto --gas-adjustment 1.2

Reviewer 2: Wallet should be some `$PROVER_ADMIN`, don't hardcode env specific names

```

> ### Setup reward pools

Reviewer 2: Can reward pools be setup before the update verifier set step (can ask Edwin)? Reorder if so to better pipeline it

Reviewer 3: seems that there is no requirement for the verifier set to be updated before the rewards pool is created. I don't think the order matters as long as both are executed, otherwise gmp will fail. @milapsheth you think is better to create the pool first? probably makes sense in mainnet so that the pools can be created in parallel before the chain is operation

Reviewer 2: Thanks, yep, should reduce latency wherever possible

| Network              | `epoch_duration` | `participation_threshold` | `rewards_per_epoch` |
| -------------------- | ---------------- | ------------------------- | ------------------- |
| **Devnet-amplifier** | `"100"`          | `["8", "10"]`             | `"100"`             |
| **Stagenet**         | `"100"`          | `["8", "10"]`             | `"100"`             |
| **Testnet**          | `"100"`          | `["8", "10"]`             | `"100"`             |
| **Mainnet**          | `TBD`            | `TBD`                     | `TBD`               |

10. Create reward pool for voting verifier

```bash
node cosmwasm/submit-proposal.js execute \
  -c Rewards \
  -t "Create pool for stellar in stellar voting verifier" \
  -d "Create pool for stellar in stellar voting verifier" \
  --runAs $RUN_AS_ACCOUNT \
  --deposit $DEPOSIT_VALUE \
  --msg "{
    \"create_pool\": {
      \"params\": {
        \"epoch_duration\": \"100\",
        \"participation_threshold\": [
          \"8\",

>           \"10\"

Reviewer 2: don't hardcode values that might be env specific

        ],

>         \"rewards_per_epoch\": \"100\"

Reviewer 2: same

      },
      \"pool_id\": {
        \"chain_name\": \"$CHAIN\",
        \"contract\": \"$VOTING_VERIFIER\"
      }
    }
  }"
```

11. Create reward pool for multisig

```bash
node cosmwasm/submit-proposal.js execute \
  -c Rewards \
  -t "Create pool for stellar in axelar multisig" \
  -d "Create pool for stellar in axelar multisig" \
  --runAs $RUN_AS_ACCOUNT \
  --deposit $DEPOSIT_VALUE \
  --msg "{
    \"create_pool\": {
      \"params\": {

>         \"epoch_duration\": \"100\",

Reviewer 2: same

        \"participation_threshold\": [
          \"8\",
          \"10\"
        ],
        \"rewards_per_epoch\": \"100\"
      },
      \"pool_id\": {
        \"chain_name\": \"$CHAIN\",
        \"contract\": \"$MULTISIG\"
      }
    }
  }"
```

> 12. Add funds to reward pools

Reviewer 2: ```suggestion
12. Add funds to reward pools from a wallet containing the reward funds `$REWARD_AMOUNT`
```

```bash
axelard tx wasm execute $REWARDS "{ \"add_rewards\": { \"pool_id\": { \"chain_name\": \"$CHAIN\", \"contract\": \"$MULTISIG\" } } }" --amount $REWARD_AMOUNT --from $WALLET

axelard tx wasm execute $REWARDS "{ \"add_rewards\": { \"pool_id\": { \"chain_name\": \"$CHAIN\", \"contract\": \"$VOTING_VERIFIER\" } } }" --amount $REWARD_AMOUNT --from $WALLET
```

> 13. Update ampd with the Stellar chain configuration.

Reviewer 2: Move this next to the other ampd cmds?

```bash
cosmwasm_contract="[\"$VOTING_VERIFIER\"]"

> type="StellarMsgVerifier"

Reviewer 2: What about the update verifier set config?

Author: is this covered by this section ?
```
10. Create genesis verifier set

```bash
axelard tx wasm execute $MULTISIG_PROVER '"update_verifier_set"' --from $PROVER_ADMIN --gas auto --gas-adjustment 1.2
```
```

Reviewer 2: No, I'm referring to the ampd config file. It needs a handler for update verifier set. See devnet ampd config for reference

# Stellar GMP Amplifier

|                | **Owner**                            |
| -------------- | ------------------------------------ |
| **Created By** | @ahramy <ahram@interoplabs.io>       |
| **Deployment** | @RiceAndMeet <steven@interoplabs.io> |

| **Network**          | **Deployment Status** | **Date** |
| -------------------- | --------------------- | -------- |
| **Devnet Amplifier** | -                     | TBD      |
| **Stagenet**         | -                     | TBD      |
| **Testnet**          | -                     | TBD      |
| **Mainnet**          | -                     | TBD      |

- [Amplifier Releases](https://github.com/axelarnetwork/axelar-amplifier/releases)
- [VotingVerifier v1.1.0](https://github.com/axelarnetwork/axelar-amplifier/releases/tag/voting-verifier-v1.1.0)
- [Gateway v1.1.1](https://github.com/axelarnetwork/axelar-amplifier/releases/tag/gateway-v1.1.1)

> - [MultisigProver v1.1.1](https://github.com/axelarnetwork/axelar-amplifier/releases/tag/multisig-prover-v1.1.1)

Reviewer 2: I find this whole deployment process to be difficult to work with. Instead, I'd like the script to be able to take in a version / commit hash, and then just retrieve the bytecode for the contract from R2. We can then dynamically search the codeId or upload it

> - [MultisigProver v1.1.1](https://github.com/axelarnetwork/axelar-amplifier/releases/tag/multisig-prover-v1.1.1)

Author: Had a sync with Ayush, we want to have a script that verifies version / commit hash of what we're deploying.
I created a ticket to track this: https://axelarnetwork.atlassian.net/browse/AXE-7529

## Background

1. These are the instructions for deploying Amplifier contracts for Stellar connection.

### Pre-requisites

- Ensure that the [External Gateway](../stellar/2025-01-Stellar-GMP-v1.0.0.md) is deployed first, as `VotingVerifier` needs the `sourceGatewayAddress` which is the External Gateway address.

## Deployment

- Create an `.env` config. `CHAIN` should be set to `stellar` for mainnet, and `stellar-2024-q4` for all other networks.

```yaml
MNEMONIC=xyz
ENV=xyz
CHAIN=xyz
```

- Confirm `VotingVerifier(v1.1.0)`, `Gateway(v1.1.1)` and `MultisigProver(v1.1.1)` contracts are already stored in `$ENV.json`

```bash
VotingVerifier(v1.1.0) -> "storeCodeProposalCodeHash": "d9412440820a51bc48bf41a77ae39cfb33101ddc6562323845627ea2042bf708"
Gateway(v1.1.1) -> "storeCodeProposalCodeHash": "2ba600ee0d162184c9387eaf6fad655f1d75db548f93e379f0565cb2042d856f"
MultisigProver(v1.1.1) -> "storeCodeProposalCodeHash": "00428ef0483f103a6e1a5853c4b29466a83e5b180cc53a00d1ff9d022bc2f03a"
```

- Add config in `$ENV.json` to deploy Amplifier contracts.

> | Network              | `serviceName` | `votingThreshold` | `signingThreshold` | `confirmationHeight` |

Author: updated

4. Set environment variables

- Network-specific environment variables: These variables need to be updated by the network.

```bash
VOTING_VERIFIER=$(cat ./axelar-chains-config/info/$ENV.json | jq ".axelar.contracts.VotingVerifier[\"$CHAIN\"].address" | tr -d '"')
GATEWAY=$(cat ./axelar-chains-config/info/$ENV.json | jq ".axelar.contracts.Gateway[\"$CHAIN\"].address" | tr -d '"')
MULTISIG_PROVER=$(cat ./axelar-chains-config/info/$ENV.json | jq ".axelar.contracts.MultisigProver[\"$CHAIN\"].address" | tr -d '"')
MULTISIG=$(cat ./axelar-chains-config/info/$ENV.json | jq .axelar.contracts.Multisig.address | tr -d '"')
REWARDS=$(cat ./axelar-chains-config/info/$ENV.json | jq .axelar.contracts.Rewards.address | tr -d '"')
```

- Gov proposal environment variables. Update these for each network

| Network              | `DEPOSIT_VALUE` | `REWARD_AMOUNT`     |
| -------------------- | --------------- | ------------------- |
| **Devnet-amplifier** | `100000000`     | `1000000uamplifier` |
| **Stagenet**         | `100000000`     | `1000000uaxl`       |
| **Testnet**          | `2000000000`    | `1000000uaxl`       |
| **Mainnet**          | `2000000000`    | `1000000uaxl`       |

```bash

> RUN_AS_ACCOUNT=[wasm deployer/governance address]

Author: @RiceAndMeet Could you confirm that we need to have `RUN_AS_ACCOUNT` for devnet-amplifier deployment? If it's required only for devnet-amplifier, we should mention it here

> RUN_AS_ACCOUNT=[wasm deployer/governance address]

Reviewer 4: yes, for devnet-amplifier, the contract creator for AxelarGateway, Verifier and MultisgProver is set to the amplifier account, not the governance account (default value).  So `RUN_AS_ACCOUNT` is needed in this case for devnet-amplifier

> RUN_AS_ACCOUNT=[wasm deployer/governance address]

Author: Thanks, updated

PROVER_ADMIN=[prover admin who is responsible for the contract's operations]
DEPOSIT_VALUE=[deposit value]
REWARD_AMOUNT=[reward amount]
```

5. Register stellar gateway at the Router

```bash
node cosmwasm/submit-proposal.js execute \
  -c Router \
  -t "Register Gateway for stellar" \
  -d "Register Gateway address for stellar at Router contract" \
  --deposit $DEPOSIT_VALUE \
  --runAs $RUN_AS_ACCOUNT \
  --msg "{
    \"register_chain\": {
      \"chain\": \"$CHAIN\",
      \"gateway_address\": \"$GATEWAY\",
      \"msg_id_format\": \"hex_tx_hash_and_event_index\"
      }
    }"

> ```

Author: updated

> 6. Update ampd with the Stellar chain configuration. Verifiers should use their own Stellar RPC node for the `http_url` in production.

Author: moved the step after all proposals pass and re-numbered

> 6. Update ampd with the Stellar chain configuration. Verifiers should use their own Stellar RPC node for the `http_url` in production.

Reviewer 2: We don't need to wait for all proposals to pass. Once the chain is added to the router, we should announce asap so that verifiers are ready by the time proposals pass, allowing us to generate the Genesis verifier set

> 6. Update ampd with the Stellar chain configuration. Verifiers should use their own Stellar RPC node for the `http_url` in production.

Author: moved back, updated.

| Network              | `http_url`                             |
| -------------------- | -------------------------------------- |
| **Devnet-amplifier** | `https://horizon-testnet.stellar.org/` |
| **Stagenet**         | `https://horizon-testnet.stellar.org/` |
| **Testnet**          | `https://horizon-testnet.stellar.org/` |
| **Mainnet**          | `https://horizon.stellar.org`          |

```bash
[[handlers]]
type="StellarMsgVerifier"
http_url=[http url]
cosmwasm_contract="$VOTING_VERIFIER"

[[handlers]]
type="StellarVerifierSetVerifier"
http_url=[http url]
cosmwasm_contract="$VOTING_VERIFIER"

> ```

Author: added the ampd command to register here. the infrastructure repo is private, should we add it here since this will be public? cc: @milapsheth

> ```

Reviewer 2: Private infra cmds shouldn't be included. Those can be stored in a Notion FAQ page for chain ops

#### Voting Verifier Reward

| Network              | `epoch_duration` | `participation_threshold` | `rewards_per_epoch` |
| -------------------- | ---------------- | ------------------------- | ------------------- |
| **Devnet-amplifier** | `"100"`          | `["8", "10"]`             | `"100"`             |
| **Stagenet**         | `"100"`          | `["7", "10"]`             | `"100"`             |
| **Testnet**          | `"100"`          | `["7", "10"]`             | `"100"`             |
| **Mainnet**          | `TBD`            | `TBD`                     | `TBD`               |

```bash
node cosmwasm/submit-proposal.js execute \
  -c Rewards \
  -t "Create pool for stellar in stellar voting verifier" \
  -d "Create pool for stellar in stellar voting verifier" \
  --runAs $RUN_AS_ACCOUNT \
  --deposit $DEPOSIT_VALUE \
  --msg "{
    \"create_pool\": {
      \"params\": {
        \"epoch_duration\": [epoch duration],
        \"participation_threshold\": [participation threshold],
        \"rewards_per_epoch\": [rewards per epoch]
      },
      \"pool_id\": {
        \"chain_name\": \"$CHAIN\",
        \"contract\": \"$VOTING_VERIFIER\"
      }
    }
  }"

> ```

Author: updated

```bash
[[handlers]]
type="StellarMsgVerifier"
http_url=[http url]
cosmwasm_contract="[\"$VOTING_VERIFIER\"]"

[[handlers]]
type="StellarVerifierSetVerifier"
http_url=[http url]
cosmwasm_contract="[\"$VOTING_VERIFIER\"]"

```

7. Register prover contract on coordinator

```bash
node cosmwasm/submit-proposal.js execute \
  -c Coordinator \
  -t "Register Multisig Prover for stellar" \
  -d "Register Multisig Prover address for stellar at Coordinator contract" \
  --deposit $DEPOSIT_VALUE \
  --runAs $RUN_AS_ACCOUNT \
  --msg "{
    \"register_prover_contract\": {
      \"chain_name\": \"$CHAIN\",
      \"new_prover_addr\": \"$MULTISIG_PROVER\"
    }
  }"
```

8. Authorize Stellar Multisig prover on Multisig

```bash
node cosmwasm/submit-proposal.js execute \
  -c Multisig \
  -t "Authorize Multisig Prover for stellar" \
  -d "Authorize Multisig Prover address for stellar at Multisig contract" \
  --runAs $RUN_AS_ACCOUNT \
  --deposit $DEPOSIT_VALUE \
  --msg "{
    \"authorize_callers\": {
      \"contracts\": {
        \"$MULTISIG_PROVER\": \"$CHAIN\"
      }
    }
  }"
```

### Setup reward pools

9. Create reward pool for voting verifier

> #### Voting Verifier Reward

Author: reference: https://www.notion.so/bright-ambert-2bd/Stellar-ITS-Release-doc-Review-Temp-196c53fccb77800f8865ca39c464e134

### Instantiate Amplifier contracts

```bash

> CONTRACT_ADMIN=[wasm contract admin address for the upgrade and migration]

Reviewer 2: add a table with the addresses per network

Author: asked @eguajardo to see the CONTRACT_ADMIN is different based on network

Reviewer 4: I believe these are the CONTRACT_ADMIN (Multisig Governance Key) on Gateway, VotingVerifier and MultisigProver

Stagnet: `axelar12qvsvse32cjyw60ztysd3v655aj5urqeup82ky`
Testnet: `axelar12f2qn005d4vl03ssjq07quz6cja72w5ukuchv7`
Mainnet: `axelar1nctnr9x0qexemeld5w7w752rmqdsqqv92dw9am`

### Instantiate Amplifier contracts

| Network              | `CONTRACT_ADMIN`                                |
| -------------------- | ----------------------------------------------- |
| **Devnet-amplifier** | `axelar1zlr7e5qf3sz7yf890rkh9tcnu87234k6k7ytd9` |
| **Stagenet**         | `axelar12qvsvse32cjyw60ztysd3v655aj5urqeup82ky` |
| **Testnet**          | `axelar12f2qn005d4vl03ssjq07quz6cja72w5ukuchv7` |
| **Mainnet**          | `axelar1nctnr9x0qexemeld5w7w752rmqdsqqv92dw9am` |

```bash
CONTRACT_ADMIN=[wasm contract admin address for the upgrade and migration based on network]
```

1. Instantiate `VotingVerifier`

```bash

> node ./cosmwasm/deploy-contract.js instantiate -c VotingVerifier -s $CHAIN --fetchCodeId --instantiate2 --admin $CONTRACT_ADMIN

Reviewer 2: Is the salt needed vs the default one (which uses the chain name)? Was stellar deployed to devnet-amplifier with a salt override?

Author: `devnet-amplifier` deployment did not use `salt` override, but by this comment: https://github.com/axelarnetwork/axelar-contract-deployments/pull/523#discussion_r1950757304

Author: I checked the script [getSalt](https://github.com/axelarnetwork/axelar-contract-deployments/blame/main/cosmwasm/utils.js#L63) and it's retrieving `chainName` by default, we don't need to have  `-s $CHAIN` anymore.

- Gov proposal environment variables. Update these for each network

| Network              | `DEPOSIT_VALUE` | `REWARD_AMOUNT`     |
| -------------------- | --------------- | ------------------- |
| **Devnet-amplifier** | `100000000`     | `1000000uamplifier` |
| **Stagenet**         | `100000000`     | `1000000uaxl`       |
| **Testnet**          | `2000000000`    | `1000000uaxl`       |

> | **Mainnet**          | `2000000000`    | `1000000uaxl`       |

Reviewer 2: 1 AXL is not sufficient for rewards. Maybe ask Ayush/Canh what value should be used for stagnet/testnet/mainnet. Rewards need to be topped up regularly. Can be updated in a follow-up PR, but let's not forget about it, cc @RiceAndMeet

Author: I left it as `TBD` for now, as we can update it while deployment

```bash
RUN_AS_ACCOUNT=[wasm deployer/governance address]

> PROVER_ADMIN=[prover admin who is responsible for the contract's operations]

Reviewer 2: add a table

Author: asked @eguajardo about the PROVER_ADMIN values

Reviewer 2: Also see the Flow contract setting by network?

- Gov proposal environment variables. Update these for each network

| Network              | `DEPOSIT_VALUE` | `REWARD_AMOUNT`     |
| -------------------- | --------------- | ------------------- |
| **Devnet-amplifier** | `100000000`     | `1000000uamplifier` |
| **Stagenet**         | `100000000`     | `1000000uaxl`       |
| **Testnet**          | `2000000000`    | `1000000uaxl`       |
| **Mainnet**          | `2000000000`    | `1000000uaxl`       |
| Network              | `PROVER_ADMIN`                                  | `DEPOSIT_VALUE` | `REWARD_AMOUNT`     |
| -------------------- | ----------------------------------------------- | --------------- | ------------------- |
| **Devnet-amplifier** | `axelar1zlr7e5qf3sz7yf890rkh9tcnu87234k6k7ytd9` | `100000000`     | `1000000uamplifier` |
| **Stagenet**         | `axelar1l7vz4m5g92kvga050vk9ycjynywdlk4zhs07dv` | `100000000`     | `1000000uaxl`       |
| **Testnet**          | `axelar17qafmnc4hrfa96cq37wg5l68sxh354pj6eky35` | `2000000000`    | `1000000uaxl`       |
| **Mainnet**          | `axelar1pczf792wf3p3xssk4dmwfxrh6hcqnrjp70danj` | `2000000000`    | `1000000uaxl`       |

```bash

> RUN_AS_ACCOUNT=[wasm deployer/governance address]

Reviewer 2: this should be set to the governanceAddress for that network. On devnet, it's a specific key

Author: @blockchainguyy Could you help me to confirm this? based on [your comment](https://github.com/axelarnetwork/axelar-contract-deployments/pull/523#discussion_r1950770682), it seems it's picked up automatically. but on devnet-amplifier it seems not. should I add it back?

Author: same comment as this: https://github.com/axelarnetwork/axelar-contract-deployments/pull/523#discussion_r1952075229
hence resolving this one.

    }"
```

> 6. Update ampd with the Stellar chain configuration.

Reviewer 2: ```suggestion
6. Update ampd with the Stellar chain configuration. Verifiers should use their own Stellar RPC node for the `http_url` in production.
```

> 6. Update ampd with the Stellar chain configuration.

Reviewer 2: Move this below with the other ampd cmds

Author: I see. updated

| Network              | `http_url`                             |
| -------------------- | -------------------------------------- |
| **Devnet-amplifier** | `https://horizon-testnet.stellar.org/` |
| **Stagenet**         | `https://horizon-testnet.stellar.org/` |
| **Testnet**          | `https://horizon-testnet.stellar.org/` |
| **Mainnet**          | `https://horizon.stellar.org`          |

```bash
[[handlers]]
http_url=[http url]
cosmwasm_contract="[\"$VOTING_VERIFIER\"]"

> type="StellarMsgVerifier"

Reviewer 2: add the type first?

Author: updated

> [[handlers]]

Reviewer 2: ```suggestion

[[handlers]]
```

Author: updated

```bash
[[handlers]]
type="StellarMsgVerifier"
http_url=[http url]

> cosmwasm_contract="[\"$VOTING_VERIFIER\"]"

Reviewer 2: ```suggestion
cosmwasm_contract="$VOTING_VERIFIER"
```

[[handlers]]
type="StellarVerifierSetVerifier"
http_url=[http url]

> cosmwasm_contract="[\"$VOTING_VERIFIER\"]"

Reviewer 2: ```suggestion
cosmwasm_contract="$VOTING_VERIFIER"
```

    }"
```

```bash

> # Verify by executing this on genesis pod

Reviewer 2: ```suggestion
```
It's an axelar query, can be done from any node. genesis pod is internal infra

  }"
```

> ### Setup reward pools

Reviewer 2: ```suggestion
```
sectioning is weird in the middle of a numbered list

9. Create reward pool for voting verifier

#### Voting Verifier Reward

| Network              | `epoch_duration` | `participation_threshold` | `rewards_per_epoch` |
| -------------------- | ---------------- | ------------------------- | ------------------- |
| **Devnet-amplifier** | `"100"`          | `["8", "10"]`             | `"100"`             |
| **Stagenet**         | `"100"`          | `["7", "10"]`             | `"100"`             |
| **Testnet**          | `"100"`          | `["7", "10"]`             | `"100"`             |

> | **Mainnet**          | `TBD`            | `TBD`                     | `TBD`               |

Reviewer 2: mainnet is still missing

Author: Those values for Mainnet is TBD, as it needs to be checked with BD or Canh. https://www.notion.so/bright-ambert-2bd/Stellar-ITS-Release-doc-Review-Temp-196c53fccb77800f8865ca39c464e134?pvs=4#196c53fccb7780dd8aa8fa324a6e91e8 Should I ask BD or Canh about this value?

Author: Referenced to [the doc](https://www.notion.so/bright-ambert-2bd/Amplifier-Contract-Params-f7fc3401a2f040438c12ba50018383eb?pvs=4) and updated

  }"
```

```bash

> # Verify by executing this on genesis pod

Reviewer 2: same everywhere. Think a bit carefully about what these terms mean when using it

Author: ah, I see. updated.

#### Voting Verifier Reward
#### Rewards

| Network              | `epoch_duration` | `participation_threshold` | `rewards_per_epoch` |
| -------------------- | ---------------- | ------------------------- | ------------------- |
| **Devnet-amplifier** | `"100"`          | `["8", "10"]`             | `"100"`             |
| **Stagenet**         | `"100"`          | `["7", "10"]`             | `"100"`             |
| **Testnet**          | `"100"`          | `["7", "10"]`             | `"100"`             |
| **Mainnet**          | `TBD`            | `TBD`                     | `TBD`               |
| **Devnet-amplifier** | `100`            | `["7", "10"]`             | `100`               |
| **Stagenet**         | `600`            | `["7", "10"]`             | `TBD`               |
| **Testnet**          | `14845`          | `["7", "10"]`             | `TBD`               |
| **Mainnet**          | `14845`          | `["8", "10"]`             | `TBD`               |

```bash
node cosmwasm/submit-proposal.js execute \
  -c Rewards \
  -t "Create pool for stellar in stellar voting verifier" \
  -d "Create pool for stellar in stellar voting verifier" \

>   --runAs $RUN_AS_ACCOUNT \

Reviewer 2: same

  }"
```

10. Create reward pool for multisig

> #### MultiSig Reward

Reviewer 2: ```suggestion
#### Multisig Reward
```

| Network              | `epoch_duration` | `participation_threshold` | `rewards_per_epoch` |
| -------------------- | ---------------- | ------------------------- | ------------------- |
| **Devnet-amplifier** | `"100"`          | `["8", "10"]`             | `"100"`             |

> | **Stagenet**         | `"100"`          | `["8", "10"]`             | `"100"`             |

Reviewer 2: participation threshold is off vs stagenet above. double check these values for flow on each network, you can see the values on different networks on axelarscan
https://devnet-amplifier.axelarscan.io/amplifier-rewards/flow

Reviewer 2: Not sure why we have these misc inconsistencies. Let's use Flow as reference, since the original params were decided for that, unless there's a good reason for an exception
cc @blockchainguyy

Author: Referenced to [the doc](https://www.notion.so/bright-ambert-2bd/Amplifier-Contract-Params-f7fc3401a2f040438c12ba50018383eb?pvs=4) and updated

  }"
```

> 11. Add funds to reward pools from a wallet containing the reward funds `$REWARD_AMOUNT`

Reviewer 2: More funds need to be added later if we just start with 1 AXL

Reviewer 2: @RiceAndMeet Please get clarity from BD parallely, so we don't have issues later in the onboarding process

Author: This task/comment will be tracked in a separate Notion doc, hence resolving this comment.

  }"
```

10. Add funds to reward pools from a wallet containing the reward funds `$REWARD_AMOUNT`

```bash
axelard tx wasm execute $REWARDS "{ \"add_rewards\": { \"pool_id\": { \"chain_name\": \"$CHAIN\", \"contract\": \"$MULTISIG\" } } }" --amount $REWARD_AMOUNT --from $WALLET

axelard tx wasm execute $REWARDS "{ \"add_rewards\": { \"pool_id\": { \"chain_name\": \"$CHAIN\", \"contract\": \"$VOTING_VERIFIER\" } } }" --amount $REWARD_AMOUNT --from $WALLET
```

> 11. Update ampd with the Stellar chain configuration. Verifiers should use their own Stellar RPC node for the `http_url` in production.

Reviewer 2: Don't wait for all proposals to pass

Author: ok, moved back. reference: https://github.com/axelarnetwork/axelar-contract-deployments/pull/523#discussion_r1951356902

| Network              | `http_url`                             |
| -------------------- | -------------------------------------- |
| **Devnet-amplifier** | `https://horizon-testnet.stellar.org/` |
| **Stagenet**         | `https://horizon-testnet.stellar.org/` |
| **Testnet**          | `https://horizon-testnet.stellar.org/` |
| **Mainnet**          | `https://horizon.stellar.org`          |

```bash
[[handlers]]
type="StellarMsgVerifier"
http_url=[http url]
cosmwasm_contract="$VOTING_VERIFIER"

[[handlers]]
type="StellarVerifierSetVerifier"
http_url=[http url]
cosmwasm_contract="$VOTING_VERIFIER"
```

> 12. Update ampd with the Stellar chain configuration.

Reviewer 2: same

Author: ok, moved back. reference: https://github.com/axelarnetwork/axelar-contract-deployments/pull/523#discussion_r1951356902
